src/kernel/src/net/mod.rs

```diff
@@ -77,6 +78,45 @@ impl NetManager {
         todo!()
     }
 
+    fn sys_netcontrol(self: &Arc<Self>, td: &VThread, i: &SysIn) -> Result<SysOut, SysErr> {
+        let fd: i32 = i.args[0].try_into().unwrap();
+        let op: i32 = i.args[1].try_into().unwrap();
+        let buf: *mut u8 = i.args[2].into();
+        let buflen: u32 = i.args[3].try_into().unwrap();
+
+        info!("Netcontrol called with op = {op}.");
+
+        let buf = if buf.is_null() {
+            None
+        } else {
+            if buflen > 160 {
+                return Err(SysErr::Raw(EINVAL));
+            }
+
+            Some(unsafe { core::slice::from_raw_parts_mut(buf, buflen as usize) })
```

Comment:
And later, there is a similar check for copyout. But since we take the original ref and don't copy anything, I don't know what to do with that.

Reply:
It's the same thing as sys_ioctl

Reply:
> And later, there is a similar check for copyout. But since we take the original ref and don't copy anything, I don't know what to do with that.

The problem here is our behavior is difference than the PS4. On the PS4 if `0x30000000` does not set on `op` it will not read the input buffer at all. That mean it will ignore any input but our implementation is not.

Reply:
Refactored. Check it out.